### PR TITLE
[Haskell] match (--*) in punctuation.definition.comment.haskell

### PR DESCRIPTION
**Pull request by moodmosaic**
/cc @FichteFoll 

Before

image

After

image

### CODE REVIEW

File: Haskell/Haskell.sublime-syntax [hunk 1]
```diff
@@ -331,7 +331,7 @@ contexts:
   comment:
     # As of build 4079 the [:punct:] class is missing some ASCII chars that
     # Unicode considers part of the Symbol category.
-    - match: '--(?![[:punct:]!-/:-@\[-`{-~])'
+    - match: '--*(?![[:punct:]!-/:-@\[-`{-~])'
```
Comment (FichteFoll) [inline Haskell/Haskell.sublime-syntax:334]:
```suggestion
    - match: '--+(?![[:punct:]!-/:-@\[-`{-~])'
```

**Review by FichteFoll — CHANGES_REQUESTED** [review]

File: Haskell/Haskell.sublime-syntax (hunk 1 above)
Comment (moodmosaic) [reply Haskell/Haskell.sublime-syntax:334]: Ah, good catch; we want *1* or more. Thank you :+1:

**Comment by moodmosaic** [comment]
Should be good to go! Thank you @FichteFoll for the review.

File: Haskell/syntax_test_haskell.hs [hunk 2]
```diff
@@ -3,6 +3,9 @@
 23*36  -- single line comment
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+23*36  --------------------------------------------------- single line comment
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
```
Comment (deathaxe) [inline Haskell/syntax_test_haskell.hs:8]: The `^` markers should propably span the whole comment (token to test).

Comment (moodmosaic) [reply Haskell/syntax_test_haskell.hs:8]: Could you submit this as a code change suggestion?

Comment (deathaxe) [reply Haskell/syntax_test_haskell.hs:8]: The followng snippet contains a bit more accurate tests and more than requested, but I'd use ...

```
23*36  --------------------------------------------------- single line comment
--    ^ - comment - punctuation
--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
--                                                        ^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell - punctuation
```

... to also ensure boundaries are correct.

Comment (moodmosaic) [reply Haskell/syntax_test_haskell.hs:8]: Is this what you mean? If so, happy to apply the change and push it. Tests all passing:

```diff
@@ -4,10 +4,12 @@
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
 23*36  --------------------------------------------------- single line comment
 --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+--    ^ - comment - punctuation
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
 23*36
 -- <- - comment.line.double-dash.haskell
 
    {- block comment -} 23*36
 -- ^^ punctuation.definition.comment.begin.haskell
```

```
Success: 1270 assertions in 1 files passed
[Finished]
```

Comment (moodmosaic) [reply Haskell/syntax_test_haskell.hs:8]: Or, perhaps this is all we need:

```diff
@@ -2,12 +2,11 @@
 
 23*36  -- single line comment
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
 23*36  --------------------------------------------------- single line comment
---     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
---     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
 23*36
 -- <- - comment.line.double-dash.haskell
 
    {- block comment -} 23*36
 -- ^^ punctuation.definition.comment.begin.haskell
```

Comment (deathaxe) [reply Haskell/syntax_test_haskell.hs:8]: The 3 lines of my post replace those red lines in your last one.

Comment (moodmosaic) [reply Haskell/syntax_test_haskell.hs]: Done. Thank you.

**Review by deathaxe — APPROVED** [review]

**Review by FichteFoll — APPROVED** [review]

**Pull request merged**